File: client-libraries/javascript/src/resources/health.js
import Resource from './base';

/**
 * Health resource.
 */
class HealthResource extends Resource {
  /**
   * Get health status of the system, cameras, and storage.
   * @returns {Promise<Object>} Health status
   */
  async get() {
    return this.get('/api/v1/health');
  }

  /**
   * Get system health.
   * @returns {Promise<Object>} System health
   */
  async getSystem() {
    return this.get('/api/v1/health/system');
  }

  /**
   * Get camera health.
   * @returns {Promise<Object>} Camera health
   */
  async getCameras() {
    return this.get('/api/v1/health/cameras');
  }

  /**
   * Get storage health.
   * @returns {Promise<Object>} Storage health
   */
  async getStorage() {
    return this.get('/api/v1/health/storage');
  }

  /**
   * Get system statistics.
   * @returns {Promise<Object>} System statistics
   */
  async getStats() {
    return this.get('/api/v1/stats');
  }

  /**
   * Get system statistics history.
   * @param {string} keys - Comma-separated list of keys to include
   * @returns {Promise<Object>} System statistics history
   */
  async getStatsHistory(keys) {
    const params = {};
    if (keys) {
      params.keys = keys;
    }
    return this.get('/api/v1/stats/history', { params });
  }

  /**
   * Get system metrics.
   * @returns {Promise<Object>} System metrics
   */
  async getMetrics() {
    return this.get('/api/v1/metrics');
  }

  /**
   * Get ffprobe information.
   * @param {string} paths - Comma-separated list of paths to probe
   * @returns {Promise<Object>} FFprobe information
   */
  async getFfprobe(paths) {
    const params = {};
    if (paths) {
      params.paths = paths;
    }
    return this.get('/api/v1/ffprobe', { params });
  }

  /**
   * Get VA-API information.
   * @returns {Promise<Object>} VA-API information
   */
  async getVainfo() {
    return this.get('/api/v1/vainfo');
  }

  /**
   * Get NVIDIA information.
   * @returns {Promise<Object>} NVIDIA information
   */
  async getNvinfo() {
    return this.get('/api/v1/nvinfo');
  }

  /**
   * Get logs for a service.
   * @param {string} service - Service name (arkos, nginx, go2rtc)
   * @param {Object} options - Options
   * @param {string} options.download - Download filename
   * @param {boolean} options.stream - Stream logs
   * @param {number} options.start - Start line
   * @param {number} options.end - End line
   * @returns {Promise<string>} Logs
   */
  async getLogs(service, options = {}) {
    return this.get(`/api/v1/logs/${service}`, { params: options });
  }
}

export default HealthResource;

File: client-libraries/python/arkos_client/resources/health.py
"""
Health resource for the Arkos client library.
"""

from .base import Resource


class HealthResource(Resource):
    """Health resource."""

    def get(self):
        """
        Get health status of the system, cameras, and storage.

        Returns:
            dict: Health status
        """
        return self._request("GET", "/api/v1/health")

    def get_system(self):
        """
        Get system health.

        Returns:
            dict: System health
        """
        return self._request("GET", "/api/v1/health/system")

    def get_cameras(self):
        """
        Get camera health.

        Returns:
            dict: Camera health
        """
        return self._request("GET", "/api/v1/health/cameras")

    def get_storage(self):
        """
        Get storage health.

        Returns:
            dict: Storage health
        """
        return self._request("GET", "/api/v1/health/storage")

    def get_stats(self):
        """
        Get system statistics.

        Returns:
            dict: System statistics
        """
        return self._request("GET", "/api/v1/stats")

    def get_stats_history(self, keys=None):
        """
        Get system statistics history.

        Args:
            keys (str, optional): Comma-separated list of keys to include

        Returns:
            dict: System statistics history
        """
        params = {}
        if keys:
            params["keys"] = keys
        return self._request("GET", "/api/v1/stats/history", params=params)

    def get_metrics(self):
        """
        Get system metrics.

        Returns:
            dict: System metrics
        """
        return self._request("GET", "/api/v1/metrics")

    def get_ffprobe(self, paths=None):
        """
        Get ffprobe information.

        Args:
            paths (str, optional): Comma-separated list of paths to probe

        Returns:
            dict: FFprobe information
        """
        params = {}
        if paths:
            params["paths"] = paths
        return self._request("GET", "/api/v1/ffprobe", params=params)

    def get_vainfo(self):
        """
        Get VA-API information.

        Returns:
            dict: VA-API information
        """
        return self._request("GET", "/api/v1/vainfo")

    def get_nvinfo(self):
        """
        Get NVIDIA information.

        Returns:
            dict: NVIDIA information
        """
        return self._request("GET", "/api/v1/nvinfo")

    def get_logs(self, service, download=None, stream=False, start=0, end=None):
        """
        Get logs for a service.

        Args:
            service (str): Service name (arkos, nginx, go2rtc)
            download (str, optional): Download filename
            stream (bool, optional): Stream logs
            start (int, optional): Start line
            end (int, optional): End line

        Returns:
            str: Logs
        """
        params = {"stream": stream, "start": start}
        if download:
            params["download"] = download
        if end is not None:
            params["end"] = end
        return self._request("GET", f"/api/v1/logs/{service}", params=params)
